src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -673,7 +673,7 @@ public Thread newThread(Runnable r) {
         HttpConfiguration config = connector.getConnectionFactory(HttpConnectionFactory.class).getHttpConfiguration();
         // use a bigger buffer as Stapler traces can get pretty large on deeply nested URL
         config.setRequestHeaderSize(12 * 1024);
-        connector.setHost("localhost");
+        connector.setHost(System.getProperty("host", "localhost"));
```

Comment:
Seems like this would be bad, it opens up a potentially insecure instance on public interfaces, which is a drive by owning of your machine